vault/identity_store_oidc.go

```diff
@@ -747,13 +789,19 @@ func (i *IdentityStore) pathOIDCGenerateToken(ctx context.Context, req *logical.
 		return nil, err
 	}
 
+	expiry := role.TokenTTL
+	if expiry > key.VerificationTTL {
+		expiry := key.VerificationTTL
+		i.Logger().Warn("a role's token ttl cannot be longer than the verification_ttl of the key it references, setting token ttl to %d", expiry)
```

Comment:
Do we want to return the warning to the user? If so, we can add it to the return's `logical.Response` through [AddWarning](https://github.com/hashicorp/vault/blob/af5338b4852227cee9358d1b197aafe1db65e028/sdk/logical/response.go#L77).